### ISSUE TITLE
Click (touch) inside items doesn't work on iOS

### ISSUE THREAD

**Issue by undestroyer** [kenwheeler/slick#1015]
Hi.
I'm trying to use Slick to display offers. Every offer has a button, that can be clicked (touched). 
If  property `mobileFirst` set to `false` - on desktop everything works fine, but on iOS 7 Safari and Chrome jQuery `$('orderButton').on('click' function(){})` doesn't work.
If  property `mobileFirst` set to `true` - iOS starts work with click event, but desktop borwsers don't catch this.
```javascript
$(document).ready(function() { 
    var SLICKconfig = {
        infinite: true,
        slidesToShow: 3,
        slidesToScroll: 1,
        autoplay: true,
        mobileFirst:false,
        autoplaySpeed: 5000,
        prevArrow:"<div class='visible-lg vv-slick-prev'><img src='/img/arrow.png'></div>",
        nextArrow:"<div class='visible-lg vv-slick-next'><img src='/img/arrow.png' class='rotate'></div>",
        responsive: [
            {
              breakpoint: 1200,
              settings: {
                slidesToShow: 3,
                slidesToScroll: 1,
              }
            },
            {
              breakpoint: 1024,
              settings: {
                slidesToShow: 2,
                slidesToScroll: 1
              }
            },
            {
              breakpoint: 768,
              settings: {
                slidesToShow: 1,
                slidesToScroll: 1
              }
            }
        ],
    };
    $(".carousel-contaner").slick(SLICKconfig);
});
```
```javascipt
$(".hot-tour-button").on('click',function(){
        toggleShadow();
    });
```
How should I configure Slick to make click work both, on desktop and mobile devices?

**Comment by tomaszpoliszuk** [comment]
please, jsfiddle

**Comment by undestroyer** [comment]
codepen (as like as jsfiddle) examle (http://codepen.io/anon/pen/jEYymW) crashes on my iPad mini iOS7.1.2 in both browsers (Chrome and Safari) after first scrolling (and I have no idea why it happens)
You can check live demo at http://vvtur.ru . Try tap green buttons in slider items, it should show you modal form. On desktop it is OK, but on mobile it fails

**Comment by undestroyer** [comment]
Ok, little investigation helped me to get some info. Problem is in `responsive` property. I get this bug when `mobileFirst:false` number of visible elements differs from the number in `responsive.settings.slidesToShow` property. Unsetting `responsive` property makes thing work good, but it provides design problems

**Comment by tomaszpoliszuk** [comment]
@undestroyer - sorry i'm too sick right now to analyse this problem, but i'm also experiencing some problems with responsive options (didn't have time to check it also) - you could try 1.3.15 if it works, unless you need some options from 1.4.
P.S. by jsfiddle i mean any code playground (like "google that" === "check in search engine" or very old "walkman" === "cassette player [not necessary made by sony]").

**Comment by desandro** [comment]
You might want to try [Flickity](http://flickity.metafizzy.co). Flickity has a [custom `staticClick` event](http://flickity.metafizzy.co/api.html#staticclick) to resolve this tricky situation detecting clicks/taps within carousels. Try the demo: http://codepen.io/desandro/pen/XJZgzM

**Comment by neverthemachineforever** [comment]
:+1: I am having a similar issue with not being able to click anchors within a slide if the responsive property is set, in fact no events on anchors seem to fire. I'll try and confirm @undestroyer's fix, or otherwise force it to work somehow and post back.

**Comment by neverthemachineforever** [comment]
Yup, can confirm that @undestroyer's fix works as long as the event handlers are registered after slick is (re)initialised, but it's obviously not ideal to re-slick every resize.

**Comment by neverthemachineforever** [comment]
It seems that Slick calls `init()` after the resize event calls `checkResponsive()` which removes all event handlers on the slides, probably somewhere around here: https://github.com/kenwheeler/slick/blob/1.4.1/slick/slick.js#L709

My current solution is to listen for the `init` event and rebind any event handlers (Marionette in this case):
```javascript
this.$childViewContainer.on('init', function() {
  console.log('delegating child events');
  self.children.each(function(child) {
    child.delegateEvents();
  });
});
```

**Comment by mikerogerz** [comment]
My solution was to use event delegation (jquery's ".on()" method) from outside of the carousel's top element (with the "slick-slider" class), targeting the elements inside the carousel. This way the event handler is unaffected due to the bug.

**Comment by neverthemachineforever** [comment]
Also a good solution @mikerogerz 

I had some issues with infinite looping of the slider as the "cloned" slides don't retain the events, so ended up setting data- attributes on the slide anchors so I could detect which one had been clicked.

@tomaszpoliszuk, @kenwheeler I would suggest these valid workarounds are probably enough to close this issue.

**Comment by mufaddalmw** [comment]
@mikerogerz Thanks for the solution.

**Comment by klihelp** [comment]
For mikerogerz's solution need some additional thinking if you add anchors dynamically.
```$('.carousel-container').on(..) vs $(document)```

**Comment by Jakobud** [comment]
@mikerogerz solution works perfectly. In my experience it doesn't even appear that I need to bind this to a container element. I just use this and it works perfectly:

```
$('.slick-slider').on('click', '.slick-slide', function(){
  // This click event will not trigger when you drag the carousel
})
```
